src/core/Table/Table.tsx

```diff
@@ -264,8 +281,17 @@ const BaseTable = <TColumns extends readonly TableColumn[]>(
     }
   }
 
+  const tableCaptionId = `${id}-caption`;
+  const getTableCaptionId = () => (caption ? `${id}-caption` : ariaLabelledBy);
```

Comment:
Could/should this refer to the `tableCaptionId` on the previous line to avoid duplicate code?

Reply:
Check